frontend/src/metabase/databases/components/DatabaseEngineField/DatabaseEngineWidget.tsx

```diff
@@ -110,6 +116,22 @@ const EngineSearch = ({
     [options, isExpanded, isSearching, searchText],
   );
 
+  useEffect(() => {
+    if (!searchText && visibleOptions.length === 0) {
+      setDbOptionsAriaMessage("");
+      return;
+    }
+
+    const message = visibleOptions.length
+      ? `${visibleOptions.length} ${visibleOptions.length === 1 ? "result" : "results"} found. Use arrow keys to navigate.`
+      : "No results found.";
+
+    setDbOptionsAriaMessage("");
+    const timeout = setTimeout(() => setDbOptionsAriaMessage(message), 50);
```

Comment:
What's the purpose of this timeout?  I think you're effectively getting a debounce here? I'd suggest using the `useDebounce` hook  from `react-use` which makes this easier to understand as a debouncing operation.